src/main/java/org/broadinstitute/hellbender/tools/coveragemodel/math/RobustBrentSolver.java

```diff
@@ -0,0 +1,439 @@
+        final double[] fSearchGrid = new double[xSearchGrid.length];
+        for (int k = 0; k < xSearchGrid.length; k++) {
+            fSearchGrid[k] = objFunc.value(xSearchGrid[k]);
+            evaluations++;
+        }
+
+        final List<Bracket> bracketsList = detectBrackets(xSearchGrid, fSearchGrid);
+        if (bracketsList.isEmpty()) {
+            throw new NoBracketingException(min, max, fSearchGrid[0], fSearchGrid[fSearchGrid.length-1]);
+        } else {
+            final List<Double> roots = new ArrayList<>(bracketsList.size());
+            for (final Bracket bracket : bracketsList) {
+                final BrentSolverWithInitialFuncEvals solver = new BrentSolverWithInitialFuncEvals(relativeAccuracy,
+                        absoluteAccuracy, functionValueAccuracy);
+                try {
+                    final double x0 = 0.5 * (bracket.xMin + bracket.xMax);
+                    final double x = solver.solve(maxEval, objFunc, bracket.xMin, bracket.xMax, x0,
+                            bracket.fMin, bracket.fMax, objFunc.value(x0));
+                    roots.add(x);
+                } catch (final TooManyEvaluationsException | NoBracketingException ex) {
+                    /* nothing to do */
+                }
+                evaluations += solver.getEvaluations();
+            }
+            if (roots.isEmpty()) {
+                throw new NoBracketingException(min, max, fSearchGrid[0], fSearchGrid[fSearchGrid.length-1]);
+            } else if (roots.size() == 1) {
+                return roots.get(0);
+            } else {
+                /* calculate merits */
+                if (meritFunc != null) {
+                    final List<Double> merits = roots.stream().mapToDouble(meritFunc::value).boxed().collect(Collectors.toList());
+                    evaluations += merits.size();
+                    double maxMerit = merits.get(0);
+                    int maxMeritIndex = 0;
+                    for (int rootIndex = 1; rootIndex < roots.size(); rootIndex++) {
+                        if (merits.get(rootIndex) > maxMerit) {
+                            maxMerit = merits.get(rootIndex);
+                            maxMeritIndex = rootIndex;
+                        }
+                    }
+                    return roots.get(maxMeritIndex);
+                } else {
+                    switch (meritPolicy) {
+                        case SMALLEST_ROOT:
+                            return Collections.min(roots);
+                        case LARGEST_ROOT:
+                            return Collections.max(roots);
+                        default:
+                            return Double.NaN;
+                    }
+                }
+            }
+        }
+    }
+
+    public int getEvaluations() {
+        return evaluations;
+    }
+
+    @VisibleForTesting
+    double[] createFractalSearchGrid(final double min, final double max, final int numBisections,
```

Comment:
they are properly documented now. The grid works as intended: it is not supposed to be symmetric as we discussed.